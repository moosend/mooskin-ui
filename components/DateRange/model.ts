import { DateRangePickerProps } from 'react-date-range';
import { IBoxComponentProps } from '../Box/model';
import { IInputComponentProps } from '../Input/model';

export interface IDateRangePickerComponentProps extends DateRangePickerProps {
	inputProps?: IInputComponentProps;
	pickerWrapperProps?: IBoxComponentProps;
	wrapperProps?: IBoxComponentProps;
}

export interface IRangeSelection {
	endDate: Date;
	key: string;
	startDate: Date;
}
