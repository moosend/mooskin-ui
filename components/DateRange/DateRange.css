@import '../_utils/globals/screens.css';
@import '../_utils/globals/variables.css';
@import url('https://code.ionicframework.com/ionicons/2.0.1/css/ionicons.min.css');
@import url('https://fonts.googleapis.com/css?family=Alegreya+Sans:500|Convergence');

.dateRange{
    display: flex;
    position: relative;
    flex-grow: 1;
    flex-shrink: 1;
}

.wrapper{
    display: flex;
    flex-grow: 1;
    flex-shrink: 1;
    flex-direction: column;
    color: var(--inputFontColor);
}

.label{
    display: flex;
    flex-direction: column;
    justify-content: center;
    color: #6b6b6b;
     /* font-size: 13px;  */
     font-size: .92857142857rem; 
    cursor: pointer;
    align-self: flex-start;
    padding-top: 15px;
    flex-basis: 150px;
}

.description{
    display: block;
    color: var(--italicDescription);
    font-size: .78571428571rem;
    margin-top: 10px;
}

.cover{
    position: fixed;
    z-index: -1;
    top: 0px;
    right: 0px;
    bottom: 0px;
    left: 0px;
}

.icon {
    color: #bebebe;
    font-size: 22px;
    border: 0;
    width: 22px;
    height: 22px;
    padding: 0px;
    margin: 0px;
    position: absolute;
    right: 15px;
    top: 15px;
    cursor: pointer;
    user-select: none;
}

.dateInput{
    outline: none;
    padding: 15px;
    font-family: Arial,sans-serif;
    border: 0;
    color: var(--inputFontColor);
    /* font-size: 16px; */
    font-size: 1rem; 
    border-radius: 8px;
    background: #efefef;
    flex-grow: 1;
    flex-shrink: 1;
    cursor: pointer;
}

.disabled{
    cursor: not-allowed;
    color: #bfbfbf;
    background: #f6f6f6;
}

.rangeCalendar{
    position: absolute;
    top: 50px;
    left: 0;
    z-index: 99;
    box-shadow: 1px 10px 20px rgba(0,0,0,.2);
}

.dateRangeSelect div[class*=alternateContainer]{
    box-shadow: none;
    border-radius: 0px;
}

.buttonContainer {
    padding: 10px;
    background: #fff;
    display: flex;
    justify-content: flex-end;
    border-top: 1px solid #bebebe;
}

:global(.daterange-component .m-input-moment){
    background-color: #ffffff;
    user-select: none;
    /* height: 360px; */
    padding: 0;
    width: 300px;
    border-radius: 0;
    border: 0;
}

:global(.daterange-component .m-input-moment .tabs){
    margin: 0;
    /* height: 285px; */
    padding: 0 !important;
}

:global(.m-input-moment .tabs .m-calendar){
    height: unset;
}

:global(.m-input-moment .tabs .m-time){
    height: unset;
    padding-top: 0;
}

:global(.m-time.is-active){
    display: flex !important;
    flex-direction: column-reverse;
}

:global(.hideTime .m-input-moment .options) {
    display: none !important;
}

:global(.daterange-component .m-calendar tbody td){
    border: none;
    border-radius: 3px;
}

:global(.daterange-component .m-calendar tbody td:hover){
    background: #5ccdde;
}

:global(.daterange-component .m-calendar .toolbar){
    background: #5ccdde;
    display: flex;
    justify-content: space-between;
    align-items: center;
    margin-bottom: 0;
    height: 45px;
}

:global(.daterange-component .m-calendar .toolbar button){
    border: none;
    background: transparent;
    margin: 0 5px;
    height: 100%;
}

:global(.daterange-component .m-calendar .toolbar button > *){
    font-size: 30px;
    display: flex;
    align-items: center;
    justify-content: center;
}

:global(.daterange-component .m-calendar thead td){
    color: #5ccdde;
    border: none;
}

:global(.daterange-component .m-calendar table){
    margin-top: 6px;
}


:global(.daterange-component .m-calendar .toolbar .current-date){
    color: #fff;
    text-transform: uppercase;
}

:global(.daterange-component .m-input-moment .btn-save){
    display: none;
}

:global(.daterange-component .m-time .time){
    background: #5ccdde;
}

:global(.daterange-component .m-time .separater){
    color: #5ccdde;
}

:global(.daterange-component .u-slider-time .value){
    background: #fff;
}

:global(.daterange-component .u-slider-time){
    background: #000;
    height: 3px;
    width: 60%;
    align-self: center;
}

:global(.daterange-component .u-slider-time .handle:after){
    border: none;
    top: -3px;
    width: 10px;
    height: 10px;
    left: -5px;
}

:global(.daterange-component .m-time .time-text){
    text-transform: uppercase;
    color: #fff;
    margin: 8px 0;
    left: 0;
    width: 40%;
}

:global(.daterange-component .m-calendar .current-day){
    color: #5ccdde;
}

:global(.daterange-component .m-calendar .current-day):hover{
    color: #fff;
}

:global(.daterange-component .showtime){
    padding: 40px 0 30px;
}

:global(.m-time .sliders){
    background: #5ccdde;
    padding: 10px 15px !important;
    display: flex;
    flex-wrap: wrap;
}

:global(.daterange-component .m-input-moment .options){
    margin: 0;
    border-bottom: 5px solid #EDECED;
    margin-bottom: -5px;
}

:global(.daterange-component .m-input-moment .options button.is-active){
    background: #EDECED;
    color: #666;
}

:global(.daterange-component .m-input-moment .options button){
    border: 0;
    color: #C3C3C3;
    margin: 0px;
    border-radius: 0;
    height: 40px;
    font-family: 'Convergence', 'Alegreya Sans', sans-serif;
}

:global(.daterange-component .m-input-moment .options button::before){
    content: '';
    margin-left: 0;
    transform: translate(0, 2px);
}

:global(.daterange-component .m-input-moment .options .ion-calendar::before){
    content: url('../../assets/images/date/calendar_date_icon_faded.png');
}

:global(.daterange-component .m-input-moment .options button.is-active.ion-calendar::before){
    content: url('../../assets/images/date/calendar_date_icon.png');
}

:global(.daterange-component .m-input-moment .options .ion-clock::before){
    content: url('../../assets/images/date/calendar_time_icon_faded.png');
}

:global(.daterange-component .m-input-moment .options button.is-active.ion-clock::before){
    content: url('../../assets/images/date/calendar_time_icon.png');
}