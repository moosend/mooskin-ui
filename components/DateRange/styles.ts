import styled from 'styled-components';

// Models
import { IBoxComponentProps } from '../Box/model';

// Components
import { Box } from '../Box/Box';

// "CSS" variables
import variables from '../_utils/globals/variables';

export const StyledDateRange = styled(Box)<IBoxComponentProps>`
	top: 40px;
	left: 0;
	position: absolute;
	z-index: 1;
	.rdrDateRangePickerWrapper *,
	.rdrDayNumber span,
	select,
	select option {
		color: ${(props) => props.palette?.fontColors.text || variables.fontColors.text} !important;
	}
	.rdrDay:not(.rdrDayPassive) .rdrInRange ~ .rdrDayNumber span,
	.rdrDay:not(.rdrDayPassive) .rdrStartEdge ~ .rdrDayNumber span,
	.rdrDay:not(.rdrDayPassive) .rdrEndEdge ~ .rdrDayNumber span,
	.rdrDay:not(.rdrDayPassive) .rdrSelected ~ .rdrDayNumber span {
		color: ${(props) => props.palette?.fontColors.white || variables.fontColors.white} !important;
	}
	.rdrStartEdge,
	.rdrInRange,
	.rdrEndEdge,
	.rdrDayToday .rdrDayNumber span:after {
		background-color: ${(props) => props.palette?.backgroundColors.primary1 || variables.backgroundColors.primary1} !important;
	}
	.rdrNextPrevButton,
	.rdrDateDisplayWrapper {
		background-color: ${(props) => props.palette?.backgroundColors.background || variables.backgroundColors.background} !important;
	}
	.rdrDateDisplayItemActive,
	.rdrDayEndPreview,
	.rdrDayStartPreview,
	.rdrDayInPreview {
		border-color: ${(props) => props.palette?.borderColors.primary1 || variables.borderColors.primary1} !important;
	}
	.rdrDayPassive {
		color: ${(props) => props.palette?.fontColors.medgray1 || variables.fontColors.medgray1} !important;
	}
	.rdrDefinedRangesWrapper,
	.rdrStaticRange,
	.rdrInputRangeInput {
		border-color: ${(props) => props.palette?.borderColors.gray1 || variables.borderColors.gray1} !important;
	}
	.rdrDayToday:not(.rdrDayPassive) .rdrInRange ~ .rdrDayNumber span:after,
	.rdrDayToday:not(.rdrDayPassive) .rdrStartEdge ~ .rdrDayNumber span:after,
	.rdrDayToday:not(.rdrDayPassive) .rdrEndEdge ~ .rdrDayNumber span:after,
	.rdrDayToday:not(.rdrDayPassive) .rdrSelected ~ .rdrDayNumber span:after,
	.rdrDefinedRangesWrapper,
	.rdrStaticRange,
	.rdrInputRangeInput,
	.rdrDateDisplayItem,
	.rdrCalendarWrapper {
		background-color: ${(props) => props.palette?.backgroundColors.white || variables.backgroundColors.white} !important;
	}
`;

StyledDateRange.displayName = 'StyledDateRange';
