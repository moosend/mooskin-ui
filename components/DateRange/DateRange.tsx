import * as React from 'react';

// Mooskin Context HoC that passes context to component props
import { withMooskinContext } from '../Styled/MooskinContextProvider';

// Date Range Component
import { DateRangePicker } from 'react-date-range';
import 'react-date-range/dist/styles.css';
import 'react-date-range/dist/theme/default.css';

// Helpers
import format from 'date-fns/format';

// Models
import { IBoxComponentProps } from '../Box/model';
import { IDateRangePickerComponentProps } from './model';

// Components
import { Box } from '../Box/Box';
import { Input, InputContainer } from '../Input/Input';

import { StyledDateRange } from './styles';

const defaultFormat = 'dd MMM yyyy';

/**
 * DateRange
 */
export const DateRange: React.FC<IDateRangePickerComponentProps> = withMooskinContext(
	({ className = '', direction = 'horizontal', format: dateFormat = defaultFormat, ...props }) => {
		const [showPicker, setShowPicker] = React.useState(false);

		const getInputValue = () => {
			let inputValue = '';

			props.ranges &&
				props.ranges.length &&
				props.ranges.forEach((item, i) => {
					if (item.startDate && item.endDate) {
						const separator = i > 0 ? ', ' : '';
						inputValue = inputValue + separator + `${format(item.startDate, dateFormat)} - ${format(item.endDate, dateFormat)}`;
					}
				});

			return inputValue ? inputValue : 'N/A';
		};

		return (
			<Box position="relative" d="flex" {...props.wrapperProps}>
				<InputContainer value={getInputValue()} {...props.inputContainerPrpps}>
					{props.customComponent}
					<Input onFocus={() => setShowPicker(true)} {...props.inputProps} />
				</InputContainer>
				{showPicker && (
					<StyledDateRange boxShadow="md" {...props.pickerWrapperProps} palette={(props as any).palette}>
						{React.createElement(DateRangePicker as React.ComponentType<any>, {
							...props,
							locale: props.locale,
							ranges: props.ranges,
							onChange: props.onChange
						})}
						<DateRangeOverlay onClick={() => setShowPicker(false)} />
					</StyledDateRange>
				)}
			</Box>
		);
	}
);

DateRange.displayName = 'DateRange';

/**
 * DateRangeOverlay
 */
const DateRangeOverlay: React.FC<IBoxComponentProps> = ({ className = '', style = {}, ...props }) => {
	return <Box position="fixed" top={0} left={0} right={0} bottom={0} zIndex={-1} {...props} />;
};

DateRangeOverlay.displayName = 'DateRangeOverlay';
