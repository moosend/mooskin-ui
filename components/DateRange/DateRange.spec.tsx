import * as React from 'react';
import { DateRange } from './DateRange';

import { mount } from 'enzyme';

const fn = jest.fn();

describe('DateRange', () => {
	test('renders correctly', () => {
		const mockDate = new Date(1466424490000);
		const tree = mount(
			<DateRange
				onChange={fn}
				ranges={[
					{
						endDate: mockDate,
						key: 'selection',
						startDate: mockDate,
					},
				]}
			/>
		);
		expect(tree).toMatchSnapshot();
	});
});
