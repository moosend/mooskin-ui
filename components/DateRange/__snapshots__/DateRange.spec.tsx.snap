// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateRange renders correctly 1`] = `
<DateRange
  className=""
  direction="horizontal"
  format="dd MMM yyyy"
  onChange={[MockFunction]}
  ranges={
    Array [
      Object {
        "endDate": 2016-06-20T12:08:10.000Z,
        "key": "selection",
        "startDate": 2016-06-20T12:08:10.000Z,
      },
    ]
  }
>
  <Component
    className=""
    direction="horizontal"
    format="dd MMM yyyy"
    onChange={[MockFunction]}
    palette={
      Object {
        "backgroundColors": Object {
          "background": "#FAFAFA",
          "black": "#000000",
          "darkgray1": "#777777",
          "darkgray2": "#454545",
          "gray1": "#E0E0E0",
          "gray2": "#E7E7E7",
          "medgray1": "#757575",
          "medgray2": "#888888",
          "primary1": "#4169E1",
          "primary2": "#212121",
          "primary3": "#00796B",
          "primary4": "#D32F2F",
          "secondary1": "#2E7D32",
          "secondary2": "#757575",
          "secondary3": "#673AB7",
          "secondary4": "#9C27B0",
          "secondary5": "#00796B",
          "secondary6": "#ECF0FC",
          "text": "#212121",
          "white": "#FFFFFF",
        },
        "borderColors": Object {
          "background": "#FAFAFA",
          "black": "#000000",
          "darkgray1": "#777777",
          "darkgray2": "#454545",
          "gray1": "#E0E0E0",
          "gray2": "#E7E7E7",
          "medgray1": "#757575",
          "medgray2": "#888888",
          "primary1": "#4169E1",
          "primary2": "#212121",
          "primary3": "#00796B",
          "primary4": "#D32F2F",
          "secondary1": "#2E7D32",
          "secondary2": "#757575",
          "secondary3": "#673AB7",
          "secondary4": "#9C27B0",
          "secondary5": "#00796B",
          "secondary6": "#ECF0FC",
          "text": "#212121",
          "white": "#FFFFFF",
        },
        "fontColors": Object {
          "background": "#FAFAFA",
          "black": "#000000",
          "darkgray1": "#777777",
          "darkgray2": "#454545",
          "gray1": "#E0E0E0",
          "gray2": "#E7E7E7",
          "medgray1": "#757575",
          "medgray2": "#888888",
          "primary1": "#4169E1",
          "primary2": "#212121",
          "primary3": "#00796B",
          "primary4": "#D32F2F",
          "secondary1": "#2E7D32",
          "secondary2": "#757575",
          "secondary3": "#673AB7",
          "secondary4": "#9C27B0",
          "secondary5": "#00796B",
          "secondary6": "#ECF0FC",
          "text": "#212121",
          "white": "#FFFFFF",
        },
      }
    }
    ranges={
      Array [
        Object {
          "endDate": 2016-06-20T12:08:10.000Z,
          "key": "selection",
          "startDate": 2016-06-20T12:08:10.000Z,
        },
      ]
    }
  >
    <Box
      className=""
      d="flex"
      position="relative"
      style={Object {}}
      transition="0.2s all"
    >
      <Component
        className=""
        d="flex"
        palette={
          Object {
            "backgroundColors": Object {
              "background": "#FAFAFA",
              "black": "#000000",
              "darkgray1": "#777777",
              "darkgray2": "#454545",
              "gray1": "#E0E0E0",
              "gray2": "#E7E7E7",
              "medgray1": "#757575",
              "medgray2": "#888888",
              "primary1": "#4169E1",
              "primary2": "#212121",
              "primary3": "#00796B",
              "primary4": "#D32F2F",
              "secondary1": "#2E7D32",
              "secondary2": "#757575",
              "secondary3": "#673AB7",
              "secondary4": "#9C27B0",
              "secondary5": "#00796B",
              "secondary6": "#ECF0FC",
              "text": "#212121",
              "white": "#FFFFFF",
            },
            "borderColors": Object {
              "background": "#FAFAFA",
              "black": "#000000",
              "darkgray1": "#777777",
              "darkgray2": "#454545",
              "gray1": "#E0E0E0",
              "gray2": "#E7E7E7",
              "medgray1": "#757575",
              "medgray2": "#888888",
              "primary1": "#4169E1",
              "primary2": "#212121",
              "primary3": "#00796B",
              "primary4": "#D32F2F",
              "secondary1": "#2E7D32",
              "secondary2": "#757575",
              "secondary3": "#673AB7",
              "secondary4": "#9C27B0",
              "secondary5": "#00796B",
              "secondary6": "#ECF0FC",
              "text": "#212121",
              "white": "#FFFFFF",
            },
            "fontColors": Object {
              "background": "#FAFAFA",
              "black": "#000000",
              "darkgray1": "#777777",
              "darkgray2": "#454545",
              "gray1": "#E0E0E0",
              "gray2": "#E7E7E7",
              "medgray1": "#757575",
              "medgray2": "#888888",
              "primary1": "#4169E1",
              "primary2": "#212121",
              "primary3": "#00796B",
              "primary4": "#D32F2F",
              "secondary1": "#2E7D32",
              "secondary2": "#757575",
              "secondary3": "#673AB7",
              "secondary4": "#9C27B0",
              "secondary5": "#00796B",
              "secondary6": "#ECF0FC",
              "text": "#212121",
              "white": "#FFFFFF",
            },
          }
        }
        position="relative"
        style={Object {}}
        transition="0.2s all"
      >
        <StyledBox
          className=""
          d="flex"
          palette={
            Object {
              "backgroundColors": Object {
                "background": "#FAFAFA",
                "black": "#000000",
                "darkgray1": "#777777",
                "darkgray2": "#454545",
                "gray1": "#E0E0E0",
                "gray2": "#E7E7E7",
                "medgray1": "#757575",
                "medgray2": "#888888",
                "primary1": "#4169E1",
                "primary2": "#212121",
                "primary3": "#00796B",
                "primary4": "#D32F2F",
                "secondary1": "#2E7D32",
                "secondary2": "#757575",
                "secondary3": "#673AB7",
                "secondary4": "#9C27B0",
                "secondary5": "#00796B",
                "secondary6": "#ECF0FC",
                "text": "#212121",
                "white": "#FFFFFF",
              },
              "borderColors": Object {
                "background": "#FAFAFA",
                "black": "#000000",
                "darkgray1": "#777777",
                "darkgray2": "#454545",
                "gray1": "#E0E0E0",
                "gray2": "#E7E7E7",
                "medgray1": "#757575",
                "medgray2": "#888888",
                "primary1": "#4169E1",
                "primary2": "#212121",
                "primary3": "#00796B",
                "primary4": "#D32F2F",
                "secondary1": "#2E7D32",
                "secondary2": "#757575",
                "secondary3": "#673AB7",
                "secondary4": "#9C27B0",
                "secondary5": "#00796B",
                "secondary6": "#ECF0FC",
                "text": "#212121",
                "white": "#FFFFFF",
              },
              "fontColors": Object {
                "background": "#FAFAFA",
                "black": "#000000",
                "darkgray1": "#777777",
                "darkgray2": "#454545",
                "gray1": "#E0E0E0",
                "gray2": "#E7E7E7",
                "medgray1": "#757575",
                "medgray2": "#888888",
                "primary1": "#4169E1",
                "primary2": "#212121",
                "primary3": "#00796B",
                "primary4": "#D32F2F",
                "secondary1": "#2E7D32",
                "secondary2": "#757575",
                "secondary3": "#673AB7",
                "secondary4": "#9C27B0",
                "secondary5": "#00796B",
                "secondary6": "#ECF0FC",
                "text": "#212121",
                "white": "#FFFFFF",
              },
            }
          }
          position="relative"
          style={Object {}}
          transition="0.2s all"
        >
          <div
            className="sc-bdnylx cFVWAa"
            d="flex"
            style={Object {}}
          >
            <Input
              className=""
              onFocus={[Function]}
              style={Object {}}
              value="20 Jun 2016 - 20 Jun 2016"
            >
              <Component
                className=""
                onFocus={[Function]}
                palette={
                  Object {
                    "backgroundColors": Object {
                      "background": "#FAFAFA",
                      "black": "#000000",
                      "darkgray1": "#777777",
                      "darkgray2": "#454545",
                      "gray1": "#E0E0E0",
                      "gray2": "#E7E7E7",
                      "medgray1": "#757575",
                      "medgray2": "#888888",
                      "primary1": "#4169E1",
                      "primary2": "#212121",
                      "primary3": "#00796B",
                      "primary4": "#D32F2F",
                      "secondary1": "#2E7D32",
                      "secondary2": "#757575",
                      "secondary3": "#673AB7",
                      "secondary4": "#9C27B0",
                      "secondary5": "#00796B",
                      "secondary6": "#ECF0FC",
                      "text": "#212121",
                      "white": "#FFFFFF",
                    },
                    "borderColors": Object {
                      "background": "#FAFAFA",
                      "black": "#000000",
                      "darkgray1": "#777777",
                      "darkgray2": "#454545",
                      "gray1": "#E0E0E0",
                      "gray2": "#E7E7E7",
                      "medgray1": "#757575",
                      "medgray2": "#888888",
                      "primary1": "#4169E1",
                      "primary2": "#212121",
                      "primary3": "#00796B",
                      "primary4": "#D32F2F",
                      "secondary1": "#2E7D32",
                      "secondary2": "#757575",
                      "secondary3": "#673AB7",
                      "secondary4": "#9C27B0",
                      "secondary5": "#00796B",
                      "secondary6": "#ECF0FC",
                      "text": "#212121",
                      "white": "#FFFFFF",
                    },
                    "fontColors": Object {
                      "background": "#FAFAFA",
                      "black": "#000000",
                      "darkgray1": "#777777",
                      "darkgray2": "#454545",
                      "gray1": "#E0E0E0",
                      "gray2": "#E7E7E7",
                      "medgray1": "#757575",
                      "medgray2": "#888888",
                      "primary1": "#4169E1",
                      "primary2": "#212121",
                      "primary3": "#00796B",
                      "primary4": "#D32F2F",
                      "secondary1": "#2E7D32",
                      "secondary2": "#757575",
                      "secondary3": "#673AB7",
                      "secondary4": "#9C27B0",
                      "secondary5": "#00796B",
                      "secondary6": "#ECF0FC",
                      "text": "#212121",
                      "white": "#FFFFFF",
                    },
                  }
                }
                style={Object {}}
                value="20 Jun 2016 - 20 Jun 2016"
              >
                <StyledInputSolo
                  boxAs="input"
                  className=""
                  onFocus={[Function]}
                  palette={
                    Object {
                      "backgroundColors": Object {
                        "background": "#FAFAFA",
                        "black": "#000000",
                        "darkgray1": "#777777",
                        "darkgray2": "#454545",
                        "gray1": "#E0E0E0",
                        "gray2": "#E7E7E7",
                        "medgray1": "#757575",
                        "medgray2": "#888888",
                        "primary1": "#4169E1",
                        "primary2": "#212121",
                        "primary3": "#00796B",
                        "primary4": "#D32F2F",
                        "secondary1": "#2E7D32",
                        "secondary2": "#757575",
                        "secondary3": "#673AB7",
                        "secondary4": "#9C27B0",
                        "secondary5": "#00796B",
                        "secondary6": "#ECF0FC",
                        "text": "#212121",
                        "white": "#FFFFFF",
                      },
                      "borderColors": Object {
                        "background": "#FAFAFA",
                        "black": "#000000",
                        "darkgray1": "#777777",
                        "darkgray2": "#454545",
                        "gray1": "#E0E0E0",
                        "gray2": "#E7E7E7",
                        "medgray1": "#757575",
                        "medgray2": "#888888",
                        "primary1": "#4169E1",
                        "primary2": "#212121",
                        "primary3": "#00796B",
                        "primary4": "#D32F2F",
                        "secondary1": "#2E7D32",
                        "secondary2": "#757575",
                        "secondary3": "#673AB7",
                        "secondary4": "#9C27B0",
                        "secondary5": "#00796B",
                        "secondary6": "#ECF0FC",
                        "text": "#212121",
                        "white": "#FFFFFF",
                      },
                      "fontColors": Object {
                        "background": "#FAFAFA",
                        "black": "#000000",
                        "darkgray1": "#777777",
                        "darkgray2": "#454545",
                        "gray1": "#E0E0E0",
                        "gray2": "#E7E7E7",
                        "medgray1": "#757575",
                        "medgray2": "#888888",
                        "primary1": "#4169E1",
                        "primary2": "#212121",
                        "primary3": "#00796B",
                        "primary4": "#D32F2F",
                        "secondary1": "#2E7D32",
                        "secondary2": "#757575",
                        "secondary3": "#673AB7",
                        "secondary4": "#9C27B0",
                        "secondary5": "#00796B",
                        "secondary6": "#ECF0FC",
                        "text": "#212121",
                        "white": "#FFFFFF",
                      },
                    }
                  }
                  style={Object {}}
                  value="20 Jun 2016 - 20 Jun 2016"
                >
                  <Box
                    boxAs="input"
                    className="sc-eCApGN sc-jSFkmK fQlSyR kGAihP"
                    onFocus={[Function]}
                    palette={
                      Object {
                        "backgroundColors": Object {
                          "background": "#FAFAFA",
                          "black": "#000000",
                          "darkgray1": "#777777",
                          "darkgray2": "#454545",
                          "gray1": "#E0E0E0",
                          "gray2": "#E7E7E7",
                          "medgray1": "#757575",
                          "medgray2": "#888888",
                          "primary1": "#4169E1",
                          "primary2": "#212121",
                          "primary3": "#00796B",
                          "primary4": "#D32F2F",
                          "secondary1": "#2E7D32",
                          "secondary2": "#757575",
                          "secondary3": "#673AB7",
                          "secondary4": "#9C27B0",
                          "secondary5": "#00796B",
                          "secondary6": "#ECF0FC",
                          "text": "#212121",
                          "white": "#FFFFFF",
                        },
                        "borderColors": Object {
                          "background": "#FAFAFA",
                          "black": "#000000",
                          "darkgray1": "#777777",
                          "darkgray2": "#454545",
                          "gray1": "#E0E0E0",
                          "gray2": "#E7E7E7",
                          "medgray1": "#757575",
                          "medgray2": "#888888",
                          "primary1": "#4169E1",
                          "primary2": "#212121",
                          "primary3": "#00796B",
                          "primary4": "#D32F2F",
                          "secondary1": "#2E7D32",
                          "secondary2": "#757575",
                          "secondary3": "#673AB7",
                          "secondary4": "#9C27B0",
                          "secondary5": "#00796B",
                          "secondary6": "#ECF0FC",
                          "text": "#212121",
                          "white": "#FFFFFF",
                        },
                        "fontColors": Object {
                          "background": "#FAFAFA",
                          "black": "#000000",
                          "darkgray1": "#777777",
                          "darkgray2": "#454545",
                          "gray1": "#E0E0E0",
                          "gray2": "#E7E7E7",
                          "medgray1": "#757575",
                          "medgray2": "#888888",
                          "primary1": "#4169E1",
                          "primary2": "#212121",
                          "primary3": "#00796B",
                          "primary4": "#D32F2F",
                          "secondary1": "#2E7D32",
                          "secondary2": "#757575",
                          "secondary3": "#673AB7",
                          "secondary4": "#9C27B0",
                          "secondary5": "#00796B",
                          "secondary6": "#ECF0FC",
                          "text": "#212121",
                          "white": "#FFFFFF",
                        },
                      }
                    }
                    style={Object {}}
                    transition="0.2s all"
                    value="20 Jun 2016 - 20 Jun 2016"
                  >
                    <Component
                      boxAs="input"
                      className="sc-eCApGN sc-jSFkmK fQlSyR kGAihP"
                      onFocus={[Function]}
                      palette={
                        Object {
                          "backgroundColors": Object {
                            "background": "#FAFAFA",
                            "black": "#000000",
                            "darkgray1": "#777777",
                            "darkgray2": "#454545",
                            "gray1": "#E0E0E0",
                            "gray2": "#E7E7E7",
                            "medgray1": "#757575",
                            "medgray2": "#888888",
                            "primary1": "#4169E1",
                            "primary2": "#212121",
                            "primary3": "#00796B",
                            "primary4": "#D32F2F",
                            "secondary1": "#2E7D32",
                            "secondary2": "#757575",
                            "secondary3": "#673AB7",
                            "secondary4": "#9C27B0",
                            "secondary5": "#00796B",
                            "secondary6": "#ECF0FC",
                            "text": "#212121",
                            "white": "#FFFFFF",
                          },
                          "borderColors": Object {
                            "background": "#FAFAFA",
                            "black": "#000000",
                            "darkgray1": "#777777",
                            "darkgray2": "#454545",
                            "gray1": "#E0E0E0",
                            "gray2": "#E7E7E7",
                            "medgray1": "#757575",
                            "medgray2": "#888888",
                            "primary1": "#4169E1",
                            "primary2": "#212121",
                            "primary3": "#00796B",
                            "primary4": "#D32F2F",
                            "secondary1": "#2E7D32",
                            "secondary2": "#757575",
                            "secondary3": "#673AB7",
                            "secondary4": "#9C27B0",
                            "secondary5": "#00796B",
                            "secondary6": "#ECF0FC",
                            "text": "#212121",
                            "white": "#FFFFFF",
                          },
                          "fontColors": Object {
                            "background": "#FAFAFA",
                            "black": "#000000",
                            "darkgray1": "#777777",
                            "darkgray2": "#454545",
                            "gray1": "#E0E0E0",
                            "gray2": "#E7E7E7",
                            "medgray1": "#757575",
                            "medgray2": "#888888",
                            "primary1": "#4169E1",
                            "primary2": "#212121",
                            "primary3": "#00796B",
                            "primary4": "#D32F2F",
                            "secondary1": "#2E7D32",
                            "secondary2": "#757575",
                            "secondary3": "#673AB7",
                            "secondary4": "#9C27B0",
                            "secondary5": "#00796B",
                            "secondary6": "#ECF0FC",
                            "text": "#212121",
                            "white": "#FFFFFF",
                          },
                        }
                      }
                      style={Object {}}
                      transition="0.2s all"
                      value="20 Jun 2016 - 20 Jun 2016"
                    >
                      <StyledBox
                        as="input"
                        boxAs="input"
                        className="sc-eCApGN sc-jSFkmK fQlSyR kGAihP"
                        onFocus={[Function]}
                        palette={
                          Object {
                            "backgroundColors": Object {
                              "background": "#FAFAFA",
                              "black": "#000000",
                              "darkgray1": "#777777",
                              "darkgray2": "#454545",
                              "gray1": "#E0E0E0",
                              "gray2": "#E7E7E7",
                              "medgray1": "#757575",
                              "medgray2": "#888888",
                              "primary1": "#4169E1",
                              "primary2": "#212121",
                              "primary3": "#00796B",
                              "primary4": "#D32F2F",
                              "secondary1": "#2E7D32",
                              "secondary2": "#757575",
                              "secondary3": "#673AB7",
                              "secondary4": "#9C27B0",
                              "secondary5": "#00796B",
                              "secondary6": "#ECF0FC",
                              "text": "#212121",
                              "white": "#FFFFFF",
                            },
                            "borderColors": Object {
                              "background": "#FAFAFA",
                              "black": "#000000",
                              "darkgray1": "#777777",
                              "darkgray2": "#454545",
                              "gray1": "#E0E0E0",
                              "gray2": "#E7E7E7",
                              "medgray1": "#757575",
                              "medgray2": "#888888",
                              "primary1": "#4169E1",
                              "primary2": "#212121",
                              "primary3": "#00796B",
                              "primary4": "#D32F2F",
                              "secondary1": "#2E7D32",
                              "secondary2": "#757575",
                              "secondary3": "#673AB7",
                              "secondary4": "#9C27B0",
                              "secondary5": "#00796B",
                              "secondary6": "#ECF0FC",
                              "text": "#212121",
                              "white": "#FFFFFF",
                            },
                            "fontColors": Object {
                              "background": "#FAFAFA",
                              "black": "#000000",
                              "darkgray1": "#777777",
                              "darkgray2": "#454545",
                              "gray1": "#E0E0E0",
                              "gray2": "#E7E7E7",
                              "medgray1": "#757575",
                              "medgray2": "#888888",
                              "primary1": "#4169E1",
                              "primary2": "#212121",
                              "primary3": "#00796B",
                              "primary4": "#D32F2F",
                              "secondary1": "#2E7D32",
                              "secondary2": "#757575",
                              "secondary3": "#673AB7",
                              "secondary4": "#9C27B0",
                              "secondary5": "#00796B",
                              "secondary6": "#ECF0FC",
                              "text": "#212121",
                              "white": "#FFFFFF",
                            },
                          }
                        }
                        style={Object {}}
                        transition="0.2s all"
                        value="20 Jun 2016 - 20 Jun 2016"
                      >
                        <input
                          className="sc-bdnylx jknATe sc-eCApGN sc-jSFkmK fQlSyR kGAihP"
                          onFocus={[Function]}
                          style={Object {}}
                          value="20 Jun 2016 - 20 Jun 2016"
                        />
                      </StyledBox>
                    </Component>
                  </Box>
                </StyledInputSolo>
              </Component>
            </Input>
          </div>
        </StyledBox>
      </Component>
    </Box>
  </Component>
</DateRange>
`;
